api/ccapi/include/tensor_dim.h

```diff
@@ -28,8 +30,56 @@
 #else
 #define _FP16 _Float16
 #endif
+#else /* !ENABLE_FP16 */
+/* Keep the FP16 programming interface, but don't allow using it in run-time */
+#define _FP16 uint16_t
```

Comment:
This is great. It definitely can be used for refactoring the multi type handle in NNTrainer.